### ISSUE TITLE
[CY-6504] `No unused vars` false positives

### ISSUE THREAD

**Issue by MrDaedra** [codacy/codacy-eslint#3158]
Recently a lot of false positives for "no unused vars" rule has appeared in our repo. Previously, the were no or almost no violations of this rule but now 1766 of 1990 issues are related to this one.
It's quite annoying because we have to ignore a lot of warnings for almost every PR in Codacy UI.
Here is a list of dev dependencies we use:
```JSON
  "devDependencies": {
    "@angular-devkit/build-angular": "^14.2.1",
    "@angular-eslint/builder": "^14.0.3",
    "@angular-eslint/eslint-plugin": "^14.0.3",
    "@angular-eslint/eslint-plugin-template": "^14.0.3",
    "@angular-eslint/schematics": "~14.0.3",
    "@angular-eslint/template-parser": "14.0.3",
    "@angular/cli": "~14.2.1",
    "@angular/compiler-cli": "~14.2.0",
    "@schematics/angular": "~14.2.1",
    "@types/chart.js-2.9.4": "npm:@types/chart.js@^2.9.32",
    "@types/jasmine": "4.3.0",
    "@types/node": "^18.7.16",
    "@types/tinycolor2": "^1.4.3",
    "@types/uuid": "^8.3.4",
    "@typescript-eslint/eslint-plugin": "^5.36.2",
    "@typescript-eslint/parser": "^5.36.2",
    "eslint": "^8.23.0",
    "eslint-config-google": "^0.14.0",
    "jasmine-core": "~4.4.0",
    "karma": "~6.4.0",
    "karma-chrome-launcher": "~3.1.1",
    "karma-coverage": "~2.2.0",
    "karma-jasmine": "~5.1.0",
    "karma-jasmine-html-reporter": "~2.0.0",
    "karma-teamcity-reporter": "^1.1.0",
    "ng-packagr": "^14.2.0",
    "postcss": "^8.4.16",
    "postcss-scss": "^4.0.4",
    "prettier": "^2.7.1",
    "stylelint": "^14.11.0",
    "stylelint-config-standard-scss": "^5.0.0",
    "stylelint-scss": "^4.3.0",
    "typescript": "4.7.4",
    "webpack-bundle-analyzer": "^4.6.1"
  }
```

**Comment by Rafaeldsb** [comment]
I got the same errors here

**Comment by stefanvacareanu7** [comment]
Hi there,

Thanks for reaching out!

We did an [eslint update](https://github.com/codacy/codacy-eslint/pull/3183) that fix the issue on the 19th of September, but we took some time to make sure the problem is indeed resolved.

In case you notice such issues on old PRs, a re-analysis should make it work.

Keep up the good work!
Stefan
